File: python/BUILDER_COMPOSITE/BUILDER_COMPOSITE/builder/ConstructorReservacionSencilla.py
from ..Reservacion.ReservacionSencilla import ReservacionSencilla
from .ConstructorReservacion import ConstructorReservacion

class ConstructorReservacionSencilla(ConstructorReservacion):
    
    def __init__(self) -> None:
        self.reservacionSencilla = ReservacionSencilla()
    
    def construirDatosCliente(self, nombres,
                              apellidoPaterno,
                              apellidoMaterno,
                              edad,
                              tipoCliente):
        self.reservacionSencilla.datosCliente.setDatosCliente(nombres,
                              apellidoPaterno,
                              apellidoMaterno,
                              edad,
                              tipoCliente)
    
    def construirDatosCompra(self, numeroTarjeta,
                             fechaVencimiento):
        self.reservacionSencilla.datosCompra.setDatosCompra(numeroTarjeta,
                             fechaVencimiento)
    
    
    def construirDatosFuncion(self, asiento,
                              horario,
                              pelicula):
        self.reservacionSencilla.datosFuncion.setDatosFuncion(asiento,
                              horario,
                              pelicula)
    
    def obtenerReservacion(self):
        return self.reservacionSencilla

File: php/BUILDER_COMPOSITE/constructores/ConstructorReservacionSencilla.php
<?php

require 'ConstructorReservacion.php';
require './reservaciones/ReservacionSencilla.php';
class ConstructorReservacionSencilla implements ConstructorReservacion{
    public $reservacionSencilla;
    
    public function __construct() {
        $this->reservacionSencilla = new ReservacionSencilla();
    }
    
    public function construirDatosFuncion($asiento, $hora, $dia, $mes, $anio, $pelicula) {
        $this->reservacionSencilla->getDatosFuncion()->setDatosFuncion($asiento,$hora, $dia, $mes, $anio, $pelicula);
    }

    public function construirDatosCliente($nombre, $apellidoP, $apellidoM, $edad, $tipoCliente) {
        $this->reservacionSencilla->getDatosCliente()->setDatosCLiente($nombre, $apellidoP, $apellidoM, $edad, $tipoCliente);
    }

    public function construirDatosCompra($numTarjeta, $fecha) {
        $this->reservacionSencilla->getDatosCompra()->setDatosCompra($numTarjeta, $fecha);
    }
    
    public function obtenerReservacion() {
        return $this->reservacionSencilla;
    }


}
